src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -58,15 +62,29 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
                 IntersectWith(ref endState, ref this.State);
             }
 
+            // Give a chance to branches internal to try/catch to resolve.
+            // Carry forward unresolved branches.
+            RestorePending(pendingBeforeTry);
```

Comment:
Is this equivalent to saving and restoring before and after the try and each individual catch block? Since we can't jump into/out of catch blocks?

Reply:
Yes, we are relying on the fact that branches cannot go between `catch` and try.

Actually, in IL it is legal to branch from catch to corresponding try and that does not go through finally. So even if we have cases like that, we would treat it correctly here. Not that it matters since it is illegal in the language...